lib/src/taiko/protocol_instance.rs

```diff
@@ -25,19 +42,16 @@ pub fn assemble_protocol_instance(extra: &TaikoExtra, header: &Header) -> Result
         .collect();
     let deposits_hash = deposits_hash(&deposits);
     let extra_data = string_to_bytes32(&header.extra_data);
-    //   meta.difficulty = meta.blobHash ^ bytes32(block.prevrandao * b.numBlocks *
-    // block.number);
-    let block_hash = tx_list_hash;
-    let block_hash_h256: U256 = block_hash.into();
-    let prevrando = if cfg!(feature = "pos") {
-        from_ethers_h256(extra.l1_next_block.mix_hash.unwrap_or_default()).into()
+    let prevrando: ethU256 = if cfg!(feature = "pos") {
+        extra.l1_next_block.mix_hash.unwrap_or_default().0.into()
     } else {
-        from_ethers_u256(extra.l1_next_block.difficulty)
+        extra.l1_next_block.difficulty
     };
-    let difficulty = block_hash_h256
-        ^ (prevrando
-            * U256::from(header.number)
-            * U256::from(extra.l1_next_block.number.unwrap_or_default().as_u64()));
+    let difficulty = calc_difficulty(
```

Comment:
```suggestion
    use use alloy_sol_types::SolValue;
    let difficulty = keccak::keccak(
                (
                    prevrando,
                    header.number,
                    extra.l1_next_block.number,
                )
                    .abi_encode(),
```

Reply:
abi_encode is not packed encode??

Reply:
It can be easier to use the `SolValue` trait to get an ABI encoded value.

Reply:
ok, let try to switch to SolValue

Reply:
> abi_encode is not packed encode??

It also has the `abi_encode_packed` method